InternalPythonModules/android/operabrowser.py

```diff
@@ -0,0 +1,200 @@
+from org.apache.commons.codec.binary import Base64
+from org.sleuthkit.autopsy.casemodule import Case
+from org.sleuthkit.autopsy.coreutils import Logger
+from org.sleuthkit.autopsy.coreutils import MessageNotifyUtil
+from org.sleuthkit.autopsy.coreutils import AppSQLiteDB
+from org.sleuthkit.autopsy.datamodel import ContentUtils
+from org.sleuthkit.autopsy.ingest import IngestJobContext
+from org.sleuthkit.datamodel import AbstractFile
+from org.sleuthkit.datamodel import BlackboardArtifact
+from org.sleuthkit.datamodel import BlackboardAttribute
+from org.sleuthkit.datamodel import Content
+from org.sleuthkit.datamodel import TskCoreException
+from org.sleuthkit.datamodel.Blackboard import BlackboardException
+from org.sleuthkit.datamodel.blackboardutils import WebBrowserArtifactsHelper
+
+import traceback
+import general
+
+"""
+Finds the SQLite DB for Opera browser, parses the DB for Bookmarks, Cookies, Web History
+and adds artifacts to the case.
+"""
+class OperaAnalyzer(general.AndroidComponentAnalyzer):
+
+    moduleName = "Opera Parser"
+    progName = "Opera"
+    
+    def __init__(self):
+        self._logger = Logger.getLogger(self.__class__.__name__)
+
+    def analyzeCookies(self, dataSource, fileManager, context):
+            cookiesDbs = AppSQLiteDB.findAppDatabases(dataSource, "Cookies", True, "com.opera.browser")
+            for cookiesDb in cookiesDbs:
+                try:
+                    cookiesDbHelper = WebBrowserArtifactsHelper(Case.getCurrentCase().getSleuthkitCase(),
+                                        self.moduleName, cookiesDb.getDBFile())
+                    cookiesResultSet = cookiesDb.runQuery("SELECT host_key, name, value, creation_utc FROM cookies")
+                    if cookiesResultSet is not None:
+                        while cookiesResultSet.next():
+                            createTime = cookiesResultSet.getLong("creation_utc") / 1000000 - 11644473600 # Webkit time
+                            cookiesDbHelper.addWebCookie( cookiesResultSet.getString("host_key"),
+                                                        createTime,  
+                                                        cookiesResultSet.getString("name"),
+                                                        cookiesResultSet.getString("value"),
+                                                        self.progName)
+
+                except SQLException as ex:
+                    self._logger.log(Level.WARNING, "Error processing query results for Opera cookies.", ex)
+                except (TskCoreException, BlackboardException) as ex:
+                    self._logger.log(Level.WARNING, "Failed to add Opera cookie artifacts.", ex)
+                finally:      
+                    cookiesDb.close()                    
+            	
+        
+
+    def analyzeHistory(self, dataSource, fileManager, context):
+            historyDbs = AppSQLiteDB.findAppDatabases(dataSource, "History", True, "com.opera.browser")
```

Comment:
same idea with the package name, it should be a class parameter so that if it needed to change we would only need to change in one spot.